src/screens/Details/Details.navigation.tsx

```diff
@@ -0,0 +1,55 @@
+import 'react-native-gesture-handler';
+import { StackNavigationProp, createStackNavigator } from '@react-navigation/stack';
+import { ProductAdded } from './components';
+import { Details as DetailsScreen } from './Details';
+import { RouteProp } from '@react-navigation/native';
+import { ChooseColor } from './components/ChooseColor';
+import { LoginToContinue } from './components/LoginToContinue';
+
+const DetailsStack = createStackNavigator<RootSackParamList>();
+type DetailsStckProps = {
+  navigation: StackNavigationProp<RootSackParamList, 'DetailsStack'>;
+  route: RouteProp<RootSackParamList, 'DetailsStack'>
+}
+
+export function DetailStack({route, navigation}: DetailsStckProps) {
+  return (
+    <DetailsStack.Navigator
+      initialRouteName='Details'
+    >
+      <DetailsStack.Group>
+        <DetailsStack.Screen
+        name="Details"
+        component={DetailsScreen}
+        initialParams={{id: route.params.id}}
+        options={{headerShown: false}}
+      />
+      </DetailsStack.Group>
+      <DetailsStack.Group screenOptions={{
+        presentation: 'card',
```

Comment:
Why 'card' presentation is provided for the group of modals?
https://reactnavigation.org/docs/stack-navigator/#presentation